File: app/src/main/java/org/torproject/android/ui/AboutDialogFragment.java
package org.torproject.android.ui;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import org.torproject.android.R;
import org.torproject.android.core.DiskUtils;
import org.torproject.android.service.OrbotService;

import java.io.IOException;

import IPtProxy.IPtProxy;

public class AboutDialogFragment extends DialogFragment {

    public static final String TAG = AboutDialogFragment.class.getSimpleName();
    private static final String BUNDLE_KEY_TV_ABOUT_TEXT = "about_tv_txt";
    private TextView tvAbout;
    private static final String ABOUT_LICENSE_EQUALSIGN = "===============================================================================";

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View view = getActivity().getLayoutInflater().inflate(R.layout.layout_about, null);
        String version;

        try {
            version = getContext().getPackageManager().getPackageInfo(
                    getContext().getPackageName(), 0).versionName + " (Tor " +
                    OrbotService.BINARY_TOR_VERSION + ")";
        } catch (PackageManager.NameNotFoundException e) {
            version = "Version Not Found";
        }

        TextView versionName = view.findViewById(R.id.versionName);
        versionName.setText(version);

        tvAbout = view.findViewById(R.id.aboutother);

        TextView tvObfs4 = view.findViewById(R.id.tvObfs4);
        tvObfs4.setText(getString(R.string.obfs4_url, IPtProxy.obfs4ProxyVersion()));

        TextView tvSnowflake = view.findViewById(R.id.tvSnowflake);
        tvSnowflake.setText(getString(R.string.snowflake_url, IPtProxy.snowflakeVersion()));

        boolean buildAboutText = true;

        if (savedInstanceState != null) {
            String tvAboutText = savedInstanceState.getString(BUNDLE_KEY_TV_ABOUT_TEXT);
            if (tvAboutText != null) {
                buildAboutText = false;
                tvAbout.setText(tvAboutText);
            }
        }

        if (buildAboutText) {
            try {
                String aboutText = DiskUtils.readFileFromAssets("LICENSE", getContext());
                aboutText = aboutText.replaceAll(ABOUT_LICENSE_EQUALSIGN, "\n").replace("\n\n", "<br/><br/>").replace("\n", "");
                tvAbout.setText(Html.fromHtml(aboutText));
            } catch (IOException e) {
            }
        }
        return new AlertDialog.Builder(getContext(), R.style.OrbotDialogTheme)
                .setTitle(getString(R.string.button_about))
                .setView(view)
                .create();
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString(BUNDLE_KEY_TV_ABOUT_TEXT, tvAbout.getText().toString());
    }
}

File: app/src/main/java/org/torproject/android/ui/AboutDialogFragment.kt
package org.torproject.android.ui

import android.app.AlertDialog
import android.app.Dialog
import android.graphics.Typeface
import android.os.Bundle
import android.text.SpannableStringBuilder
import android.text.Spanned
import android.text.style.StyleSpan
import android.view.View
import android.widget.TextView

import androidx.fragment.app.DialogFragment

import org.torproject.android.R
import org.torproject.android.BuildConfig
import org.torproject.android.core.DiskUtils
import org.torproject.android.service.OrbotService

import java.io.IOException

import IPtProxy.IPtProxy

class AboutDialogFragment : DialogFragment() {

    companion object {
        const val TAG = "AboutDialogFragment"
        const val VERSION = BuildConfig.VERSION_NAME
        private const val BUNDLE_KEY_TV_ABOUT_TEXT = "about_tv_txt"
        private const val ABOUT_LICENSE_EQUALSIGN =
            "==============================================================================="
    }

    private lateinit var tvAbout: TextView

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val view: View? = activity?.layoutInflater?.inflate(R.layout.layout_about, null)

        val versionName = view?.findViewById<TextView>(R.id.versionName)
        versionName?.text = VERSION

        tvAbout = view?.findViewById(R.id.aboutother) as TextView

        val tvTor = view.findViewById<TextView>(R.id.tvTor)
        tvTor.text = getString(R.string.tor_url, OrbotService.BINARY_TOR_VERSION)

        val tvObfs4 = view.findViewById<TextView>(R.id.tvObfs4)
        tvObfs4.text = getString(R.string.obfs4_url, IPtProxy.lyrebirdVersion())

        val tvSnowflake = view.findViewById<TextView>(R.id.tvSnowflake)
        tvSnowflake.text = getString(R.string.snowflake_url, IPtProxy.snowflakeVersion())

        var buildAboutText = true

        savedInstanceState?.getString(BUNDLE_KEY_TV_ABOUT_TEXT)?.let {
            buildAboutText = false
            tvAbout.text = it
        }

        if (buildAboutText) {
            try {
                var aboutText = DiskUtils.readFileFromAssets("LICENSE", requireContext())
                aboutText = aboutText.replace(ABOUT_LICENSE_EQUALSIGN, "\n")

                val spannableAboutText = SpannableStringBuilder(aboutText)
                spannableAboutText.setSpan(StyleSpan(Typeface.BOLD), 0, aboutText.indexOf("\n"), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)

                tvAbout.text = spannableAboutText
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }

        return AlertDialog.Builder(context, R.style.OrbotDialogTheme)
            .setTitle(getString(R.string.button_about))
            .setView(view)
            .create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(BUNDLE_KEY_TV_ABOUT_TEXT, tvAbout.text.toString())
    }
}
